editoast/src/views/infra/mod.rs

```diff
@@ -801,6 +807,66 @@ pub async fn fetch_all_infra_states(
         .collect())
 }
 
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WaypointsForm {
+    waypoints: Vec<OperationalPointReference>,
+}
+
+/// Retrieve operational points from operational point references
+#[utoipa::path(
+    get, path = "",
+    tag = "infra",
+    params(InfraIdParam),
+    request_body = inline(WaypointsForm),
+    responses(
+        (status = 200, description = "Operational points related to operational point references", body = Vec<Vec<OperationalPoint>>)
+    ),
+)]
+async fn match_operational_points(
+    State(AppState { db_pool, .. }): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    Path(InfraIdParam { infra_id }): Path<InfraIdParam>,
+    Json(waypoints_input): Json<WaypointsForm>,
+) -> Result<Json<Vec<Vec<OperationalPoint>>>> {
+    let authorized = auth
+        .check_roles([Role::OperationalStudies].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+    let mut operational_points: Vec<Vec<OperationalPoint>> = vec![];
+    let conn = &mut db_pool.get().await?;
+    for waypoint in waypoints_input.waypoints {
+        operational_points.push(match waypoint.reference {
+            OperationalPointIdentifier::OperationalPointId {
+                operational_point: op_id,
+            } => retrieve_op_from_ids(conn, infra_id, &[op_id.0])
+                .await?
+                .iter()
+                .map(|(_, op_model)| op_model.schema.clone())
+                .collect::<Vec<_>>(),
```

Comment:
You can go a step further and avoid the `.clone()`. Since the result of `retrieve_op_from_ids()` will not be used elsewhere, you can consume it with `.into_iter()` instead of `iter()`, which will yield owned objects instead of object's reference. And therefore, you'll be able to remove the `.clone()`.

```suggestion
                .into_iter()
                .map(|(_, op_model)| op_model.schema)
                .collect::<Vec<_>>(),
```

Same reasoning for the two other similar operations below.